Common/Brokerages/TDAmeritradeBrokerageModel.cs

```diff
@@ -0,0 +1,153 @@
+
+        /// <summary>
+        /// Constructor for TDAmeritrade brokerage model
+        /// </summary>
+        /// <param name="accountType">Cash or Margin</param>
+        public TDAmeritradeBrokerageModel(AccountType accountType = AccountType.Margin) : base(accountType)
+        {
+
+        }
+
+        /// <summary>
+        /// Returns true if the brokerage could accept this order. This takes into account
+        /// order type, security type, and order size limits.
+        /// </summary>
+        /// <remarks>
+        /// For example, a brokerage may have no connectivity at certain times, or an order rate/size limit
+        /// </remarks>
+        /// <param name="security">The security of the order</param>
+        /// <param name="order">The order to be processed</param>
+        /// <param name="message">If this function returns false, a brokerage message detailing why the order may not be submitted</param>
+        /// <returns>True if the brokerage could process the order, false otherwise</returns>
+        public override bool CanSubmitOrder(Security security, Order order, out BrokerageMessageEvent message)
+        {
+            if (!IsValidOrderSize(security, order.Quantity, out message))
+            {
+                return false;
+            }
+
+            message = null;
+
+            if (!_supportSecurityTypes.Contains(security.Type))
+            {
+                message = new BrokerageMessageEvent(BrokerageMessageType.Warning, "NotSupported",
+                    StringExtensions.Invariant($"The {nameof(TDAmeritradeBrokerageModel)} does not support {security.Type} security type.")
+                );
+
+                return false;
+            }
+
+            if (!_supportOrderTypes.Contains(order.Type))
+            {
+                message = new BrokerageMessageEvent(BrokerageMessageType.Warning, "NotSupported",
+                    StringExtensions.Invariant($"{order.Type} order is not supported by TDAmeritrade. Currently, only Market Order is supported.")
+                );
+
+                return false;
+            }
+
+            return base.CanSubmitOrder(security, order, out message);
+        }
+
+        /// <summary>
+        /// TDAmeritrade support Update Order
+        /// </summary>
+        /// <param name="security">Security</param>
+        /// <param name="order">Order that should be updated</param>
+        /// <param name="request">Update request</param>
+        /// <param name="message">Outgoing message</param>
+        /// <returns>True if the brokerage would allow updating the order, false otherwise</returns>
+        public override bool CanUpdateOrder(Security security, Order order, UpdateOrderRequest request, out BrokerageMessageEvent message)
+        {
+            message = null;
+
+            // validate order quantity
+            if (request.Quantity != null)
```

Comment:
This is not accurate, we could be updating just the limit price for example. Suggest just removing this logic and just returning true, let the brokerage validate it, like we do it in other brokerage implementations